### PR TITLE
Feature test

### PR DESCRIPTION
**Pull request by XiaoPa**
- fixed "'ascii' code error;
- updated results field 'rows'

### CODE REVIEW

File: src/flask_app.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def input_data():
 	except Exception ,e:
 		status = 'error'
 		results = 'data format not recoginzed'
-		return jsonify({'status': status, 'error':str(e),'results':data}), 400
+		return jsonify({'status': status, 'error':str(e),'results':'no data in the results'}), 400
```
Comment (maZahaca) [inline src/flask_app.py]: `results` cannot be different formats
- for success [...] 
- for error "..."

It should be an array all the time, if it's presented in response.

**Review by maZahaca — CHANGES_REQUESTED** [review]

File: src/flask_app.py (hunk 1 above)
Comment (XiaoPa) [reply src/flask_app.py]: updated

File: src/flask_app.py [hunk 2]
```diff
@@ -27,7 +27,7 @@ def input_data():
 	except Exception ,e:
 		status = 'error'
 		results = 'data format not recoginzed'
-		return jsonify({'status': status, 'error':str(e),'results':'no data in the results'}), 400
+		return jsonify({'error':str(e),'status': status, 'results':"[no data in the results]"}), 400
```
Comment (maZahaca) [inline src/flask_app.py]: This is still wrong format.
- I mean if we have a error, we don't need to set results or set it as an empty array `[]`.
  - results is only available for providing results of grouping, if we didn't group anything, so it's nothing to provide
- above variable `results` is unused

**Review by maZahaca — CHANGES_REQUESTED** [review]

File: src/flask_app.py (hunk 2 above)
Comment (XiaoPa) [reply src/flask_app.py]: so I just leave results as empty array:
e.g 
jsonify({'status': status, 'error':str(e),'results':'[]'}), 400
or 
jsonify({'status': status, 'error':str(e)}), 400

Comment (maZahaca) [reply src/flask_app.py]: better `jsonify({'status': status, 'error':str(e)}), 400`

**Comment by maZahaca** [comment]
Good job, thanks

**Pull request merged**